File: 2ºP - Desenvolvimento Back-End/Desenvolvimento com C#/TPs/TP2/ex12/Jogo.cs
namespace ex12;

public class Jogo {
    public int NumeroSecreto { get; set; }
    public int Tentativas { get; set; }

    public Jogo() {
        Random random = new Random();
        NumeroSecreto = random.Next(1, 101); 
        Tentativas = 0;
    }

    public void IniciarJogo() {
        int palpite;
        bool acertou = false;

        while (!acertou) {
            Console.Write("Tente adivinhar o número entre 1 e 100: ");
            palpite = int.Parse(Console.ReadLine());
            Tentativas++;

            if (palpite < NumeroSecreto){
                Console.WriteLine("O número é maior.");
            } else if (palpite > NumeroSecreto) {
                Console.WriteLine("O número é menor.");
            } else {
                acertou = true;
                Console.WriteLine($"\nParabéns! Você acertou o número {NumeroSecreto} em {Tentativas} tentativas.");
            }
        }
    }
}

File: 2P-DesenvolvimentoBack-End/01-FundamentosDevComJava/TPs/TP2/ex09/Jogo.java
package com.codingloria.TP2.ex09;

import java.util.Random;
import java.util.Scanner;

public class Jogo {
    private int numeroSecreto;
    private int tentativas;

    public Jogo() {
        Random random = new Random();
        this.numeroSecreto = random.nextInt(100) + 1;
        this.tentativas = 0;
    }

    public void jogar() {
        Scanner sc = new Scanner(System.in);
        int palpite;

        System.out.println("Adivinhe um número entre 1 e 100!");

        do {
            System.out.print("Seu palpite: ");
            palpite = sc.nextInt();
            tentativas++;

            if (palpite < numeroSecreto) {
                System.out.println("Muito baixo! Tente novamente.");
            } else if (palpite > numeroSecreto) {
                System.out.println("Muito alto! Tente novamente.");
            }
        } while (palpite != numeroSecreto);

        System.out.println("Parabéns! Você acertou em " + tentativas + " tentativas.");
        sc.close();
    }
}
